operator/api/v1alpha1/service_types.go

```diff
@@ -25,17 +25,13 @@ import (
 
 // ServiceSpec defines the desired state of Service
 type ServiceSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
-
-	// Foo is an example field of Service. Edit service_types.go to remove/update
-	Foo string `json:"foo,omitempty"`
+	ApplicationName   string    `json:"application,omitempty"`
+	PreDeplymentCheck EventSpec `json:"preDeploymentChecks"`
```

Comment:
```suggestion
	PreDeploymentCheck EventSpec `json:"preDeploymentChecks"`
```
According to the last discussions (@thisthat), I'd go for PreDeploymentExperiments and PreDeploymentAnalysis instead of PreDeploymentChecks

Reply:
We discussed it in the team, that all the fixes will be handled in another PR, because not merging this PR will block us in other progress